.github/workflows/publish-site.yml

```diff
@@ -1,35 +1,30 @@
 name: Website Build
 on:
-  push:
-    branches:
-      - 'main'
+    push:
+        branches:
+            - "develop/3.0"
 jobs:
-  Build:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: 'true'
-        token: ${{ secrets.PUSHABLE_GITHUB_TOKEN }}
-    - name: Setup .NET 5.0
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 5.0.401
-    - name: Setup .NET 6.0
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.100
-    - name: Checkout Website Branch
-      run: |
-        git fetch --all
-        git checkout ci/github-pages
-        git config --local user.email "9011267+dotnet-bot@users.noreply.github.com"
-        git config --local user.name "The Silk.NET Automaton"
-        git merge --allow-unrelated-histories -X theirs main
-    - name: Run Statiq
-      run: dotnet run -c Release --project src/Website/Silk.NET.Statiq/Silk.NET.Statiq.csproj -- -l debug --nocache
-    - name: Push to Website Branch
-      run: |
-        git add -f docs/
-        git commit -am "Push latest built GitHub Pages site"
-        git push
+    Build:
+        environment: cf-kai-3.0-docs
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+              with:
+                  submodules: "true"
+            - name: Setup .NET 6.0
+              uses: actions/setup-dotnet@v1
+              with:
+                  dotnet-version: 6.0.201
+            - name: Run Statiq
+              run: dotnet run -c Release --project src/infrastructure/Silk.NET.Statiq/Silk.NET.Statiq.csproj -- -l debug --nocache
+            - name: Install wrangler
+              run: npm install -g wrangler@beta
+            - name: Install Deps of CF worker
+              run: npm i
+              working-directory: ./website/cf-worker/
```

Comment:
You've deleted all trace of the GitHub Pages based solution. I'd personally prefer the CloudFlare stuff to be _in addition to_ the GitHub stuff, rather than instead of. If you need ideas, I'd just copy the website into a folder matching the branch name so it's deployed to e.g. https://dotnet.github.io/Silk.NET/develop/3.0

If you need to force push, overwrite/detach history, and/or modify the main pipeline, you have my blessing.